content/extending-chat-widget/custom-identity-provider/index.mdx

```diff
@@ -0,0 +1,173 @@
+---
+weight: 60
+category: "extending-chat-widget"
+subcategory: "custom-identity-provider"
+title: "Custom Identity Provider"
+tagline: "Pair your own customer identity with LiveChat"
+desc: "Cross-platfrom chats for your applications"
+versionGroup: "chat-widget"
+---
+
+# Introduction
+
+## What is it for?
+
+If you have more advanced service, handling user sessions, you can pair LiveChat customer identity with your own. As a result, your customers will never loose chats they made with your support. If you also use LiveChat on other platfrom (e.g. in mobile application), their chats will sync making the communication even more seamless and natural.
+
+## Setup
+Customer Identity Provider is essentially a singleton function that you can define as a part of `LiveChatWidget` object in tracking code that you use to install ChatWidget on your website.
+
+```html
+<!-- Start of LiveChat (www.livechat.com) code -->
+<script>
+    window.__lc = window.__lc || {};
+    window.__lc.license = <LICENSE_NUMBER>;
+    window.__lc.custom_identity_provider = function() {
+      return {
+          getToken,
+          getFreshToken,
+          hasToken,
+          invalidate
+      }
+    }
```

Comment:
it will throw because those functions are not defined :trollface: which should be pretty obvious, its described later - it would be an overkill to code everything here and then agin in examples imo

Reply:
> it will throw because those functions are not defined

yes, that's exactly what I'm referring to - it's described **later**, without any reference here what this actually does

we know that, as we did that, but IMO it's a bit illogical flow now, where the "mystery" is solved only later on

I'm not saying about coding that fully here, but perhaps make this more explicit, why these are not defined, or just rephrase that section?

Reply:
although it says "setup", it doesn't actually say anything about the setup itself

Reply:
maybe I'm a bit biased, but I don't think ppl just go through the docs as a whole 🤷🏼‍♂️ 

I'd rather say they go step-by-step, or even just scan for the interesting bits and pieces, and follow that - here, although it looks like something already cooked to be ready, is certainly not, with a lot of gaps to fill

Reply:
Right, I could switch the order here. My aim to start from there was to imply where it should be put in the beginning but maybe its more logical to say what first and then define where